cmd/bosun/search/search.go

```diff
@@ -284,6 +284,20 @@ func (s *Search) TagKeysByMetric(Metric string) []string {
 	return r
 }
 
+func (s *Search) TagKeysByMetricTree() map[string][]string {
```

Comment:
This name does not describe to me what it actually does. Which is the top level key? I am assuming the hierarchy is `metric -> all tagks`. Perhaps `MetricsWithTagKeys` would make a bit more sense.